pkg/apis/samplesoperator/v1alpha1/types.go

```diff
@@ -22,42 +22,50 @@ type SamplesResource struct {
 	Status            SamplesResourceStatus `json:"status,omitempty"`
 }
 
+type SamplesDistributionType string
+
+const (
+	RHELSamplesDistribution   = SamplesDistributionType("rhel")
+	CentosSamplesDistribution = SamplesDistributionType("centos")
+)
+
 type SamplesResourceSpec struct {
 	// defaults to registry.redhat.io
-	SamplesRegistry string
-	// whether to install the rhel imagestreams
-	InstallRHEL bool
-	// whether to install the centos imagestreams
-	InstallCentos bool
+	SamplesRegistry string `json:"sampleRegistry,omitempty" protobuf:"bytes,1,opt,name=sampleRegistry"`
```

Comment:
part of next push

Also, what the default is needs to be tweaked, as the InstallType has bearing.

Reply:
i was thinking about how this would work when i was reviewing the PR.  I think there should be no default for this field.

if it's empty string, you don't touch the imagestreams you are loading.  (and you will either load the rhel or centos ones depending on the installtype).  the imagestreams have the correct registry baked in.

If you set this, you are saying "replace all the registry values in the imagestreams with this value".  So most people would not set it and it would be fine.  Only someone who wanted to get the imagestream content from some internal registry where they'd cloned all the images would set it.